### ISSUE TITLE
react-native-cli@2.0.0 doesn't report react-native version in project directory

### ISSUE THREAD

**Issue by jsynowiec** [facebook/react-native#11463]
### Description

After upgrading react-native-cli to 2.0 and initializing a new project using `react-native init`, running `react-native -v` in the project directory doesn't report react-native version or that this is a react-native project.

```
$ react-native -v           
react-native-cli: 2.0.0
react-native: n/a - not inside a React Native project directory
```

This affects editor extensions like vscode-react-native, see https://github.com/Microsoft/vscode-react-native/issues/365

`react-native-cli@1.3.0` reports the react-native version correctly in the directory initialized using 2.0:

```
$ npm i -g react-native-cli@1.3.0
$ react-native -v
react-native-cli: 1.3.0
react-native: 0.39.2
```

Related commit: https://github.com/facebook/react-native/commit/76cd2f7bf26c98e828da18c551ccd1c41e74a21b

### Reproduction

Install react-native-cli 2.0: `npm i -g react-native-cli@2.0.0`
Initialize a new project: `react-native init AwesomeProject`
Run `react-native -v`

### Solution

Assign `REACT_NATIVE_PACKAGE_JSON_PATH` before calling the `checkForVersionArgument`.

### Additional Information

* React Native version: 0.39.2 (cli 2.0.0)
* Platform: n/a
* Operating System: MacOS

**Comment by jsynowiec** [comment]
The problem is that the `REACT_NATIVE_PACKAGE_JSON_PATH` function is declared *after* the `checkForVersionArgument` call site so on https://github.com/facebook/react-native/blob/76cd2f7bf26c98e828da18c551ccd1c41e74a21b/react-native-cli/index.js#L322 it's undefined and the exception is thrown.

**Comment by mkonicek** [comment]
Just released react-native-cli 2.0.1 with the fix: https://www.npmjs.com/package/react-native-cli
